File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/dotnet/EventHub/SingleKafkaTriggerQueueOutput.cs
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Kafka;
using Microsoft.Azure.WebJobs.Extensions.Storage;
using Microsoft.Extensions.Logging;

namespace EventHub
{
    public class SingleKafkaTriggerQueueOutput
    {
        // KafkaTrigger sample 
        // Consume the message from "topic" on the LocalBroker.
        // For EventHubs
        [FunctionName("SingleKafkaTriggerQueueOutput")]
        [return: Queue("e2e-dotnet-single-eventhub")]
        public static string Run(
                          "e2e-kafka-dotnet-single-eventhub",
                          Username = "$ConnectionString",
                          Protocol = BrokerProtocol.SaslSsl,
                          AuthenticationMode = BrokerAuthenticationMode.Plain,
                          ConsumerGroup = "$Default")] KafkaEventData<string> kevent, ILogger log)
        {
            log.LogInformation($"C# Kafka trigger function processed a message: {kevent.Value}");
            return kevent.Value;
        }
    }
}

File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/java/EventHub/src/main/java/com/function/SingleKafkaTriggerQueueOutput.java
package com.function;

import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.*;
import org.json.*;

public class SingleKafkaTriggerQueueOutput {
    /**
     * This function consume KafkaEvents on the localhost. Change the topic, EventHubBrokerList, and consumerGroup to fit your enviornment.
     * The function is trigged one for each KafkaEvent
     * @param kafkaEventData
     * @param context
     */
    @FunctionName("SingleKafkaTriggerQueueOutput")
    public void runOne(
            @KafkaTrigger(name = "kafkaTrigger", 
                          topic = "e2e-kafka-java-single-eventhub", 
                          brokerList="EventHubBrokerList",
                          username = "$ConnectionString",
                          password = "%EventHubConnectionString%",
                          authenticationMode = BrokerAuthenticationMode.PLAIN,
                          protocol = BrokerProtocol.SASLSSL,
                          consumerGroup="$Default") String kafkaEventData,
            @QueueOutput(name = "message", 
                         queueName = "e2e-java-single-eventhub", 
                         connection = "AzureWebJobsStorage") OutputBinding<String> message,
            final ExecutionContext context) {
        context.getLogger().info(kafkaEventData);
        JSONObject kafkaEventJsonObj = new JSONObject(kafkaEventData);
        context.getLogger().info(kafkaEventJsonObj.getString("Value"));
        message.setValue(kafkaEventJsonObj.getString("Value"));
    }
}
